File: JANUARY/31-01-2026/744.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        char ans=' ';
        int lb=0, ub=letters.length-1, mid;
        while(lb<=ub){
            mid=(lb+ub)/2;
            if(letters[mid]>target){
                ans=letters[mid];
                ub=mid-1;
            }
            else{
                lb=mid+1;
            }
        }
        if(ans==' ') return letters[0];
        return ans;
    }
}

File: 2026POTD/JANUARY/31-01-2026/744.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        char ans = '#';
        int start = 0, end = letters.size() - 1;
        while (start <= end) {
            int mid = (start + end) >> 1;
            if (letters[mid] > target) {
                ans = letters[mid];
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return (ans == '#') ? letters[0] : ans;
    }
};